app/src/main/java/com/codingblocks/cbonlineapp/mycourse/MyCourseRepository.kt

```diff
@@ -80,6 +82,22 @@ class MyCourseRepository(
             }
             getSectionContent(courseSection.id, runAttempt.id, courseSection.name)
         }
+        deleteOldSections(runAttempt.run?.sections?.map { it.id }!!, runAttempt.run?.id)
+    }
+
+    /**
+     *Function to delete [SectionModel] which are no longer part of course content.
+     */
+    private suspend fun deleteOldSections(list1: List<String>, id: String?) {
+        val list2 = withContext(Dispatchers.IO) { sectionDao.getCourseSection(id!!) }
+        val sum = list1 + list2
+        val finalList = sum.groupBy { it }
```

Comment:
@championswimmer is this. okay or should i think of something else to remove section

Reply:
the comment should contain what is list1 supposed to be fed as arg into the func